catalogs/catalog-hadoop/src/main/java/com/datastrato/gravitino/catalog/hadoop/HadoopCatalogOperations.java

```diff
@@ -125,20 +168,20 @@ public void initialize(
 
   private void initAuthentication(Map<String, String> conf, Configuration hadoopConf) {
     AuthenticationConfig config = new AuthenticationConfig(conf);
-    String authType = config.getAuthType();
 
-    if (StringUtils.equalsIgnoreCase(authType, AuthenticationMethod.KERBEROS.name())) {
-      hadoopConf.set(
-          HADOOP_SECURITY_AUTHENTICATION,
-          AuthenticationMethod.KERBEROS.name().toLowerCase(Locale.ROOT));
-      UserGroupInformation.setConfiguration(hadoopConf);
-      try {
-        KerberosClient kerberosClient = new KerberosClient(conf, hadoopConf);
-        File keytabFile = kerberosClient.saveKeyTabFileFromUri(catalogInfo.id());
-        this.kerberosRealm = kerberosClient.login(keytabFile.getAbsolutePath());
-      } catch (IOException e) {
-        throw new RuntimeException("Failed to login with Kerberos", e);
-      }
+    if (config.isKerberosAuth()) {
+      this.kerberosRealm =
+          initKerberos(
+              conf, hadoopConf, NameIdentifier.of(catalogInfo.namespace(), catalogInfo.name()));
+    } else if (config.isSimpleAuth()) {
+      // TODO: change the user 'datastrato' to 'anonymous' and uncomment the following code;
+      //  uncomment the following code after the user 'datastrato' is removed from the codebase.
+      //  for more, please see https://github.com/datastrato/gravitino/issues/4013
+      // UserGroupInformation u =
+      //    UserGroupInformation.createRemoteUser(PrincipalUtils.getCurrentUserName());
+      // userInfoMap.put(
+      //    NameIdentifier.of(catalogInfo.namespace(), catalogInfo.name()),
+      //    UserInfo.of(u, false, null, null));
```

Comment:
@qqqttt123 
#4013 needs to be completed before the PR is done.

Reply:
or the following error will occur
<img width="1515" alt="image" src="https://github.com/datastrato/gravitino/assets/15794564/5d3a8090-61b8-4f65-8b88-6988d94bd930">

https://github.com/datastrato/gravitino/actions/runs/9744134755/job/26889119677?pr=3852

Reply:
@qqqttt123 
Please take a look if you have some free time.

Reply:
Is the prequisition merged?

Reply:
We have reached an agreement that sub-entities use the parent's authentication if no authentication is set. When the catalog authentication is simple, schema or fileset will use the default user `anonymous` to access HDFS. so
- Whether we need to keep these changes.
- Change the default user name in the Docker image for Hive. (NOT advised by @xunliu )


Before #4013 or something similar is resolved, I think these changes should be commented. 


> Is the prerequisite pull request merged?

Not really, @xunliu is not suggested to do so.  What about changing the default user name `anonymous` to `datastrato`?

Reply:
It's not reasonable solution for me.